python/pyarrow/tests/test_pandas.py

```diff
@@ -4082,6 +4082,18 @@ def test_array_to_pandas():
         # tm.assert_series_equal(result, expected)
 
 
+def test_roundtrip_empty_table_with_extension_dtype_index():
+    if Version(pd.__version__) < Version("1.0.0"):
+        pytest.skip("ExtensionDtype to_pandas method missing")
+
+    df = pd.DataFrame(index=pd.interval_range(start=0, end=3))
+    table = pa.table(df)
+    table.to_pandas().index == pd.Index([{'left': 0, 'right': 1},
```

Comment:
This is a different issue, but what strikes me here is that the result is an object-dtype index, and not a proper IntervalIndex (so the interval dtype is not preserved on a roundtrip, while this would be the case if it was a normal column instead of the index)

Reply:
Yes, that bothered me also (see comment in https://issues.apache.org/jira/browse/ARROW-15253).

I created a separate issue to get this corrected:
https://issues.apache.org/jira/browse/ARROW-15565